### PR TITLE
Add TestStreamIsActive

### LINKED ISSUE #285: Add tests for StreamIsActive
The scaler's current `StreamIsActive` method does not have a test. It should. This should be done as part of, or after #97, although that's not required. Any tests written could be later retro-fitted to work with the improvement in #97  

### Use-Case

Adds reliability and stability long term to the scaler.

### Specification

- At least one unit test exists that calls and tests the output of `StreamIsActive`

### PR DESCRIPTION
**Pull request by asw101**
This PR adds a test for StreamIsActive. We also change the behavior of StreamIsActive to return an error when no host is found rather than logging and continuing.
  
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [] Any necessary documentation is added, such as:
  - [`README.md`](/README.md)
  - [The `docs/` directory](./docs)
  - [The docs repo](https://github.com/kedacore/keda-docs)

Fixes #285

### CODE REVIEW

File: scaler/handlers_test.go [hunk 1]
```diff
@@ -88,9 +93,140 @@ func TestIsActive(t *testing.T) {
...
+				lggr,
+				pinger,
+				table,
+				123,
+				200,
+			)
+
+			bufSize := 1024 * 1024
+			lis := bufconn.Listen(bufSize)
+			grpcServer := grpc.NewServer()
```
Comment (arschles) [inline scaler/handlers_test.go:110]:
```suggestion
			grpcServer := grpc.NewServer()
			defer grpcServer.Stop()
```

if you do this, then likely need to remove the `log.Fatalf` below if the `grpcServer.Serve(lis)` function returns a non-nil error

**Review by arschles — CHANGES_REQUESTED** [review]
So far so good! I think we still need to add some tests for error & edge cases for `StreamIsActive`

**Review by arschles — APPROVED** [review]
Looks great, thanks @asw101! We'll do the edge case/error checking tests in a follow-up PR
